purchase_stock_bl_date/models/stock_move.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2023 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    @api.multi
+    def _get_price_unit(self):
+        self.ensure_one()
+        bl_date = self.picking_id.bl_date
+        if bl_date:
+            self = self.with_context(bl_date=bl_date)
+        return super()._get_price_unit()
+
+    def _create_account_move_line(
+        self, credit_account_id, debit_account_id, journal_id
+    ):
+        self.ensure_one()
+        bl_date = self.picking_id.bl_date
+        if bl_date:
+            self = self.with_context(force_period_date=bl_date)
+        return super()._create_account_move_line(
+            credit_account_id, debit_account_id, journal_id
+        )
```

Comment:
This is out of scope of the enhancement this time.  Although we might want to seek a way to provide an option for users to force a date in pickings, the solution should be implemented in a comprehensive manner, not only for the case where B/L date is available.

```suggestion
```